doc/examples/segment_extending_clustering_framework.py

```diff
@@ -0,0 +1,233 @@
+"""
+===========================================================
+How to make new `Feature` and new `Metric` for Quickbundles
+===========================================================
+
+This tutorial shows how to create new `Feature` and new `Metric` classes that
+can be used by QuickBundles.
+
+Clustering framework
+====================
+Dipy provides a simple, flexible and fast framework to do clustering of
+sequential data (e.g. streamlines).
+
+A *sequential datum* in Dipy is represented as a numpy array of size
+:math:`(N \times D)` where each row of the array represent a D dimensional
+point of the sequence. A set of these sequences is represented as a list of
+numpy arrays of size :math:`(N_i \times D)` for :math:`i=1:M` where $M$ is the
+number of sequences in the set.
+
+This clustering framework is modular and divided in three parts:
+1) features extraction
+2) distance computation
+3) data partitioning
+
+The features extraction part regroups any preprocessing needed to be done on
```

Comment:
feature
regroups is not clear here. Clarify please. An example of a simple feature would be helpful here.